Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs

```diff
@@ -296,6 +298,20 @@ ValueTask<HttpRequestMessage> CreateRequestMessage()
                                     string message =
                                             $"GatewayStoreClient Request Timeout. Start Time UTC:{startDateTimeUtc}; Total Duration:{(DateTime.UtcNow - startDateTimeUtc).TotalMilliseconds} Ms; Request Timeout {requestTimeout.TotalMilliseconds} Ms; Http Client Timeout:{this.httpClient.Timeout.TotalMilliseconds} Ms; Activity id: {System.Diagnostics.Trace.CorrelationManager.ActivityId};";
                                     e.Data.Add("Message", message);
+                                    
+                                    if (timeoutPolicy.ShouldThrow503OnTimeout)
+                                    {
+                                        CosmosException serviceUnavailableExcpetion = CosmosExceptionFactory.CreateServiceUnavailable(
```

Comment:
You are throwing the `ServiceUnavailableException` at the end, it should be the `CosmosException`:

```suggestion
                                        throw new CosmosExceptionFactory.CreateServiceUnavailable(
```